server/data/timelineClient.test.ts

```diff
@@ -0,0 +1,48 @@
+import nock from 'nock'
+
+import { fakerEN_GB as faker } from '@faker-js/faker'
+import config from '../config'
+import paths from '../paths/api'
+import TimelineClient from './timelineClient'
+
+import { CallConfig } from './restClient'
+import { referralHistorySystemNoteFactory } from '../testutils/factories'
+
+describe('Timeline client', () => {
+  let fakeApprovedPremisesApi: nock.Scope
+  let timelineClient: TimelineClient
+
+  const callConfig = { token: 'some-token' } as CallConfig
+
+  beforeEach(() => {
+    config.apis.approvedPremises.url = 'http://localhost:8080'
+    fakeApprovedPremisesApi = nock(config.apis.approvedPremises.url)
+    timelineClient = new TimelineClient(callConfig)
+  })
+
+  afterEach(() => {
+    if (!nock.isDone()) {
+      nock.cleanAll()
+      throw new Error('Not all nock interceptors were used!')
+    }
+    nock.abortPendingRequests()
+    nock.cleanAll()
+  })
+
+  describe('fetch', () => {
+    it('returns timeline results', async () => {
```

Comment:
Would it be overkill to have a test which reflects what happens when the application is not submitted?
(Or, is there no chance of this being called if that is the case?)

Reply:
The timelines at the moment, would only be included on pages where an application has been submitted and therefore it would have an assessmentId.

I think its a bit overkill for the moment, but if we were to place timelines on say the application pages I think we would have to look at how passing application id would work and at that point we could review scenarios where application is not submitted